### DISCUSSION TITLE
Are nested query keys serialised deterministically?

### DISCUSSION THREAD

**Discussion by abhieshekumar** [reduxjs/redux-toolkit#3052] [Q&A] [answered]
It's established that a queryArgs is sorted by `key` before being used as `queryCacheKey`. For example `{name:"John", age:30}` will be used as `{age:20, name:"John"}`.

I am trying to find if the same behaviour is guaranteed in case on nested queryKeys. For example - 
queryArg - `{name:"John",info:{isActive:false, age:20}}`

does the above arg is serialised to `{info:{isActive:false, age:20}, name:"John"}` (one level of key sorting), or it is a recursive approach, which results in cache key as `{info:{age:20, isActive:false}, name:"John"}`

Thanks in advance and a happy new year 🎉

**Comment by markerikson** [comment, marked as answer]
Looking at the implementation of `defaultSerializeQueryArgs`, looks like we only go one level deep:

```ts
export const defaultSerializeQueryArgs: SerializeQueryArgs<any> = ({
  endpointName,
  queryArgs,
}) => {
  // Sort the object keys before stringifying, to prevent useQuery({ a: 1, b: 2 }) having a different cache key than useQuery({ b: 2, a: 1 })
  return `${endpointName}(${JSON.stringify(queryArgs, (key, value) =>
    isPlainObject(value)
      ? Object.keys(value)
          .sort()
          .reduce<any>((acc, key) => {
            acc[key] = (value as any)[key]
            return acc
          }, {})
      : value
  )})`
}
```

**Reply by phryneas** [reply]
That replacer *should* be applied for the full depth.

**Reply by markerikson** [reply]
OH! Y'know, it didn't even click with me that we're passing in a `JSON.stringify()` replacer function.

**Reply by abhieshekumar** [reply]
thank you.
